pkg/util/admissioncheck/admissioncheck.go

```diff
@@ -177,6 +177,17 @@ func NewAdmissionChecks(cq *kueue.ClusterQueue) map[kueue.AdmissionCheckReferenc
 			checks[checkName] = sets.New[kueue.ResourceFlavorReference]()
 		}
 	}
+	// In case an AdmissionCheck empty, it should apply to all ResourceFlavors.
```

Comment:
This could be moved under the "else" branch only? Then we could eliminate the `if len(check) != 0 {` check?

Reply:
Oh, actually we want to cover both AdmissionCheck and AdmissionCheckStrategy cases so it makes sense to keep it here

Reply:
I think the cleanest solution would be to have a list that stores all the flavors, and modify this function in following manner:


```
if admissionCheckStrategy != nil {
    iterate over checks from admissionChecksStrategy and assign either OnFlavors or if it's empty, then assign all flavors
}

iterate over checks from admissionChecks and assign all flavors to all checks
```